api/tests/unit/organisations/test_unit_organisations_models.py

```diff
@@ -338,24 +347,125 @@ def test_subscription_get_subscription_metadata_returns_free_plan_metadata_for_n
+    """
     # Given
-    max_seats = 10
-    max_api_calls = 1000000
-    subscription = Subscription(
-        subscription_id=TRIAL_SUBSCRIPTION_ID,
+    Subscription.objects.filter(organisation=organisation).update(
+        subscription_id=subscription_id, plan=plan, max_seats=max_seats
+    )
+    organisation.subscription.refresh_from_db()
+    mocker.patch("organisations.models.is_saas", return_value=False)
+    mocker.patch("organisations.models.is_enterprise", return_value=True)
+
+    # When
+    subscription_metadata = organisation.subscription.get_subscription_metadata()
+
+    # Then
+    assert subscription_metadata.projects == expected_projects
+    assert subscription_metadata.seats == expected_seats
+
+
+@pytest.mark.parametrize(
+    "subscription_id, plan, max_seats, max_api_calls, expected_seats, "
+    "expected_api_calls, expected_projects",
+    (
+        (
+            None,
+            "free",
+            10,
+            5000000,
+            MAX_SEATS_IN_FREE_PLAN,
+            MAX_API_CALLS_IN_FREE_PLAN,
+            settings.MAX_PROJECTS_IN_FREE_PLAN,
+        ),
+        ("anything", "enterprise", 20, 5000000, 20, 5000000, None),
+        (TRIAL_SUBSCRIPTION_ID, "enterprise", 20, 5000000, 20, 5000000, None),
+    ),
+)
+def test_get_subscription_metadata_for_manually_added_enterprise_saas_licenses(
+    organisation: Organisation,
+    subscription_id: str | None,
+    plan: str,
+    max_seats: int,
+    max_api_calls: int,
+    expected_seats: int,
+    expected_api_calls: int,
+    expected_projects: int | None,
+    mocker: MockerFixture,
+) -> None:
+    """
+    Specific test to make sure that we can manually add subscriptions to
+    the saas platform and the values stored in the django database will
+    be correctly used.
+    """
+    # Given
+    Subscription.objects.filter(organisation=organisation).update(
+        subscription_id=subscription_id,
+        plan=plan,
         max_seats=max_seats,
         max_api_calls=max_api_calls,
-        payment_method=None,
     )
+    organisation.subscription.refresh_from_db()
+    mocker.patch("organisations.models.is_saas", return_value=True)
 
     # When
-    subscription_metadata = subscription.get_subscription_metadata()
+    subscription_metadata = organisation.subscription.get_subscription_metadata()
 
     # Then
-    assert subscription_metadata.seats == max_seats
-    assert subscription_metadata.api_calls == max_api_calls
-    assert subscription_metadata.projects is None
+    assert subscription_metadata.projects == expected_projects
+    assert subscription_metadata.seats == expected_seats
+    assert subscription_metadata.api_calls == expected_api_calls
+
+
+def test_get_subscription_metadata_for_self_hosted_open_source(
+    organisation: Organisation, mocker: MockerFixture
+) -> None:
+    """
+    Open source should ignore the details provided in the
+    subscription and always return the free plan metadata.
+    """
```

Comment:
So if I understand this correctly, we would want all our enterprise customers to add `./ENTERPRISE_VERSION` before merging this code into their copy, correct? Or did I misunderstand?

Reply:
Nope, that is already done in the docker builds [here](https://github.com/Flagsmith/flagsmith/blob/main/.github/workflows/platform-docker-publish-all-features-image.yml#L78) and [here](https://github.com/Flagsmith/flagsmith-ee/pull/102) (this one needs merging still - I will do that not).